src/sv-pipeline/04_variant_resolution/scripts/clean_vcf_part1.sh

```diff
@@ -27,18 +27,27 @@ zcat $vcf \
   >whitelist.txt
 
 ##convert EV integer back into string##
-zcat $vcf \
-      | awk '{print $0 "\t"}' \
-      | sed -e 's/:7'"\t"'/:RD,PE,SR'"\t"'/g' \
-      | sed -e 's/:6'"\t"'/:PE,SR'"\t"'/g' \
-      | sed -e 's/:5'"\t"'/:RD,SR'"\t"'/g' \
-      | sed -e 's/:3'"\t"'/:RD,PE'"\t"'/g' \
-      | sed -e 's/:2'"\t"'/:PE'"\t"'/g' \
-      -e 's/:4'"\t"'/:SR'"\t"'/g' \
-      -e 's/:1'"\t"'/:RD'"\t"'/g' \
-      |sed 's/'"\t"'$//g' \
-      |sed 's/ID=EV,Number=1,Type=Integer/ID=EV,Number=1,Type=String/g' \
-      | bgzip > EV.update.vcf.gz
+bcftools query -f "%CHROM\t%POS\t%REF\t%ALT\t[%EV\t]\n" $vcf > ev.tab
+cut -f1-4 ev.tab > ev.sites.tab
+cut -f5- ev.tab \
+  | sed -e 's/7/RD,PE,SR/g' -e 's/6/PE,SR/g' -e 's/5/RD,SR/g' -e 's/3/RD,PE/g' -e 's/2/PE/g' -e 's/4/SR/g' -e 's/1/RD/g' \
+  > ev.replaced.list
+paste ev.sites.tab ev.replaced.list | bgzip -c > ev.replaced.tab.gz
+tabix -s1 -b2 -e2 ev.replaced.tab.gz
+
+echo '##FORMAT=<ID=EV,Number=1,Type=String,Description="Classes of evidence supporting final genotype">' > ev_header.txt
+tabix $vcf
+bcftools annotate -x FORMAT/EV $vcf | bgzip -c > no_ev.vcf.gz
```

Comment:
Do you need to remove EV here first? I thought the second annotate command would just replace the value.

Reply:
Good question, I had to go back and double-check this. The problem is that in the original vcf, EV is type Integer, but this seems to fail below since we are assigning Strings. The only solution was to delete EV entirely.